### PR TITLE
Assignment11

### PR DESCRIPTION
**Pull request by prajjwalkumarpanzade**
Login page authentication  using JWT

### CODE REVIEW

File: Assignment10/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class UsersController < ApplicationController
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      token = encode_token(user_id: @user_id)
+      render json: {data: @user,token: token}
+    else 
+      render json:{data: @user.errors.full_messages}, status: 422
```
Comment (samu80) [inline Assignment10/users_controller.rb:8]: Give proper indentation

File: Assignment10/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    end
+  end
+  private 
+  def user_params
+    params.require(:user).permit(:user_name,:password)
+  end
+
```
Comment (samu80) [inline Assignment10/users_controller.rb]: Remove extra lines

File: Assignment10/application_controller.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+class ApplicationController < ActionController::API
+  # protect_from_forgery
+  def encode_token(payload)
+    JWT.encode(payload, Rails.application.credentials[:secret_key_base]) 
+  end
+
+  def decoded_token
+    header = request.headers['Authorization']
+    if header
+        token = header.split(" ")[1]
```
Comment (samu80) [inline Assignment10/application_controller.rb]: Give 2 spaces indentations only

File: Assignment10/users_controller.rb (hunk 1 above)
Comment (prajjwalkumarpanzade) [reply Assignment10/users_controller.rb:8]: done

File: Assignment10/users_controller.rb (hunk 2 above)
Comment (prajjwalkumarpanzade) [reply Assignment10/users_controller.rb]: done

File: Assignment10/application_controller.rb (hunk 3 above)
Comment (prajjwalkumarpanzade) [reply Assignment10/application_controller.rb]: done

**Pull request merged**